elegy/nn/batch_normalization.py

```diff
@@ -0,0 +1,198 @@
+        data_format: str = "channels_last",
+        name: tp.Optional[str] = None,
+    ):
+        """
+        Creates a BatchNormalization instance.
+        
+        Arguments:
+            axis: Integer, the axis that should be normalized
+                (typically the features axis).
+                For instance, after a `Conv2D` layer with
+                `data_format="channels_first"`,
+                set `axis=1` in `BatchNormalization`.
+            momentum: Momentum for the moving average.
+            eps: Small float added to variance to avoid dividing by zero.
+            create_offset: If True, add offset of `beta` to normalized tensor.
+                se, `beta` is ignored.
+            create_scale: If True, multiply by `gamma`.
+                If False, `gamma` is not used.
+                When the next layer is linear (also e.g. `relu`),
+                this can be disabled since the scaling
+                will be done by the next layer.
+            scale_init: Optional initializer for gain (aka create_scale). Can only be set
+                if ``create_scale=True``. By default, ``1``.
+            offset_init: Optional initializer for bias (aka offset). Can only be set
+                if ``create_offset=True``. By default, ``0``.
+            axis: Which axes to reduce over. The default (``None``) signifies that all
+                but the channel axis should be normalized. Otherwise this is a list of
+                axis indices which will have normalization statistics calculated.
+            cross_replica_axis: If not ``None``, it should be a string representing
+                the axis name over which this module is being run within a ``jax.pmap``.
+                Supplying this argument means that batch statistics are calculated
+                across all replicas on that axis.
+            data_format: The data format of the input. Can be either
+                ``channels_first``, ``channels_last``, ``N...C`` or ``NC...``. By
+                default it is ``channels_last``.
+            name: The module name.
+        """
+        super().__init__(name=name)
+
+        if not create_scale and scale_init is not None:
+            raise ValueError("Cannot set `scale_init` if `create_scale=False`")
+        if not create_offset and offset_init is not None:
+            raise ValueError("Cannot set `offset_init` if `create_offset=False`")
+
+        self.create_scale = create_scale
+        self.create_offset = create_offset
+        self.eps = eps
+        self.scale_init = scale_init or jnp.ones
+        self.offset_init = offset_init or jnp.zeros
+        self.axis = axis
+        self.cross_replica_axis = cross_replica_axis
+        self.channel_index = hk_utils.get_channel_index(data_format)
+        self.mean_ema = hk.ExponentialMovingAverage(momentum, name="mean_ema")
+        self.var_ema = hk.ExponentialMovingAverage(momentum, name="var_ema")
+
+        hk.BatchNorm
```

Comment:
@cgarciae sorry for the ignorance:
- What is this void call for? (L114)
- Generally speaking, if there is already an hk.BatchNorm module, why don't just wrap it instead of rewriting it?

Reply:
1. Forgot to erase this jejeje
2. Sadly, because we need to inherit from `elegy.Module` which will have logic inside `__call__` to get the summaries.